src/games/oware.rs

```diff
@@ -0,0 +1,247 @@
+        if self.opp_pits().any(|x| self.pits[x] > 0) {
+            self.pits[mv] > 0
+        } else if self.pl_pits().any(|x| self.can_overflow(x)) {
+            self.can_overflow(mv)
+        } else {
+            true
+        }
+    }
+
+    fn play(&mut self, mv: Self::Move) {
+        assert!(self.is_available_move(mv), "{:?} is not available on {:?}", mv, self);
+        assert!(
+            self.pits.iter().chain(self.scores.iter()).sum::<u8>() == PITS as u8 * INIT_SEEDS,
+            "{} seeds should exist",
+            PITS as u8 * INIT_SEEDS
+        );
+
+        let player = usize::from(self.next_player != Player::A);
+        if self.is_final_move() {
+            self.pl_pits().for_each(|x| {
+                self.scores[player] += self.pits[x];
+                self.pits[x] = 0;
+            })
+        } else {
+            let curr_house = if self.next_player == Player::A {
+                mv
+            } else {
+                PITS / 2 + mv
+            };
+            let mut seeds = self.pits[curr_house];
+            self.pits[curr_house] = 0;
+            let mut idx = curr_house;
+
+            while seeds > 0 {
+                idx = (idx + if (idx + 1) % PITS == curr_house { 2 } else { 1 }) % PITS;
+                seeds -= 1;
+                self.pits[idx] += 1;
+            }
+
+            if self.grand_slam(idx) {
+                return;
+            }
+
+            while match self.pits[idx] {
+                2 | 3 => self.is_in_opp_pits(idx),
+                _ => false,
+            } {
+                self.scores[player] += self.pits[idx];
+                self.pits[idx] = 0;
+                idx = (idx + PITS - 1) % PITS;
+            }
+        }
+
+        let draw = self.scores.iter().all(|&x| x == PITS as u8 * INIT_SEEDS / 2) || self.is_stalemate();
+
+        let won = self.scores[player] > PITS as u8 * INIT_SEEDS / 2;
+        self.outcome = if won {
+            Some(Outcome::WonBy(self.next_player))
+        } else if draw {
+            Some(Outcome::Draw)
+        } else {
+            None
+        };
+
+        self.next_player = self.next_player.other();
+    }
+
+    fn outcome(&self) -> Option<Outcome> {
+        self.outcome
+    }
+
+    fn can_lose_after_move() -> bool {
+        false
+    }
+}
+
+impl<const PITS: usize, const INIT_SEEDS: u8> Alternating for OwareBoard<PITS, INIT_SEEDS> {}
+
+impl<const PITS: usize, const INIT_SEEDS: u8> crate::board::BoardSymmetry<OwareBoard<PITS, INIT_SEEDS>>
+    for OwareBoard<PITS, INIT_SEEDS>
+{
+    type Symmetry = crate::symmetry::UnitSymmetry;
+    type CanonicalKey = ();
+
+    fn map(&self, _: Self::Symmetry) -> Self {
+        self.clone()
+    }
+
+    fn map_move(
+        &self,
+        _: Self::Symmetry,
+        mv: <OwareBoard<PITS, INIT_SEEDS> as Board>::Move,
+    ) -> <OwareBoard<PITS, INIT_SEEDS> as Board>::Move {
+        mv
+    }
+
+    fn canonical_key(&self) -> Self::CanonicalKey {}
+}
```

Comment:
It's a bit sad that `impl_unit_symmetry_board!` doesn't work with generic bounds yet, I'll think about improving this in the future. Don't worry about it for this PR though.